detectable/src/main/java/com/synopsys/integration/detectable/detectables/pnpm/lockfile/process/PnpmYamlTransformer.java

```diff
@@ -114,18 +126,41 @@ private String convertRawEntryToPackageId(Map.Entry<String, String> entry, PnpmL
         return String.format("/%s/%s", name, version);
     }
 
-    private NameVersion parseNameVersionFromId(String id) {
+    private Optional<NameVersion> parseNameVersionFromId(String id) {
         // ids follow format: /name/version, where name often contains slashes
-        int indexOfLastSlash = id.lastIndexOf("/");
-        String name = id.substring(1, indexOfLastSlash);
-        String version = id.substring(indexOfLastSlash + 1);
+        try {
+            int indexOfLastSlash = id.lastIndexOf("/");
+            String name = id.substring(1, indexOfLastSlash);
+            String version = id.substring(indexOfLastSlash + 1);
+            return Optional.of(new NameVersion(name, version));
+        } catch (Exception e) {
+            logger.debug(String.format("There was an issue parsing package id: %s.  This is likely an unsupported format.", id));
+            return Optional.empty();
+        }
+    }
+
+    private Optional<Dependency> buildDependencyFromPackageEntry(Map.Entry<String, PnpmPackageInfo> packageEntry) {
+        PnpmPackageInfo packageInfo = packageEntry.getValue();
+        if (packageInfo.name != null && packageInfo.version != null) {
+            return Optional.of(new Dependency(externalIdFactory.createNameVersionExternalId(Forge.NPMJS, packageInfo.name, packageInfo.version)));
+        }
```

Comment:
Is it possible to have a name, but no version? If so we should create a version less external id